robot_sort/robot_sort.py

```diff
@@ -97,16 +99,49 @@ def sort(self):
         Sort the robot's list.
         """
         # Fill this out
-        pass
+        # first set the light to on then start the loop
+        self.set_light_on()
+
+        while self.light_is_on() is True:
+
+            # to exit the loop you must turn the light off to indicate false
+            self.set_light_off()
+
+            # start checking if the move right function is true in a while loop
+            while self.can_move_right() is True:
+               # swap the item and move to the right
+                self.swap_item()
+                self.move_right()
+
+                # check if the compare_item function is returning 1 and if it is swap then move left and and drop item move to the right and then turn the light on.
+                if self.compare_item() == 1:
+                    self.swap_item()
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+                    self.set_light_on()
+
+                # otherwise move left drop it and keep going
```

Comment:
hah, this is exactly how i solved it. sure, it's not beautiful, but it WORKS! great job figuring it out my man!